QEfficient/base/pytorch_transforms.py

```diff
@@ -42,6 +44,18 @@ def apply(cls, model: nn.Module) -> Tuple[nn.Module, bool]:
         transformed = False
         for module in model.modules():
             if repl_module := cls._module_mapping.get(type(module)):
+                if isinstance(module, transformers.models.llama.modeling_llama.LlamaAttention):
+                    # Special handling for LlamaAttention
+                    if not isinstance(
+                        getattr(module, "rotary_emb", None),
+                        transformers.models.llama.modeling_llama.LlamaRotaryEmbedding,
+                    ):
+                        raise TypeError("Expected LlamaAttention.rotary_emb to be of type LlamaRotaryEmbedding")
+
+                    qeff_rotary_emb = QEffLlamaRotaryEmbedding(
+                        module.head_dim, max_position_embeddings=module.max_position_embeddings, base=module.rope_theta
+                    )
+                    setattr(module, "rotary_emb", qeff_rotary_emb)
```

Comment:
Instead of specially handling Llama only, we can make this generic. After `module.__class__ = repl_module`,

```python
if hasattr(module, "__qeff_init__"):
    module.__qeff_init__()
```

See below for how to define `__qeff_init__`.